lib/resolve.ts

```diff
@@ -38,15 +38,21 @@ export function resolveTask(
 
 	const resolvers = Resolve.getDefaultResolvers();
 
-	return Resolve.resolveBundle(bundle, resolvers)
-		.then((res: Resolve.ResolvedBundle) => {
-			task.projectType = res.projectType;
-			task.buildStream = res.tarStream;
+	return new Promise((resolve, reject) => {
+		const outStream = Resolve.resolveInput(
+			bundle,
+			resolvers,
+			task.dockerfilePath,
+		);
+		task.buildStream = outStream as tar.Pack;
+		outStream.on('resolver', r => {
+			task.projectType = r;
 			task.resolved = true;
-			return task;
-		})
-		.catch(() => {
-			task.projectType = 'Could not be detected';
-			return task;
 		});
+		outStream.on('resolved-name', name => {
+			task.dockerfilePath = name;
+		});
+		outStream.on('error', reject);
+		resolve(task);
```

Comment:
This may be called before the 'resolver' or 'resolved-name' events. Is this intended?
Also `reject` may be called later.

Reply:
The `reject` being called later is definitely an issue actually, but I think that can be handled downstream anyway, the `resolve` should definitely be called first, and these fields are only populated after the build has begun. It's a little convoluted but the only way I could think to get the streaming interface we want.

Reply:
It looks like you don't need a Promise at all.
Just add the 'resolver' and 'resolved-name' event listeners.
You can still add task.buildStream.on('error', ...) later.

Reply:
or keep the Promise and resolve instead of doing `task.resolved = true`

Reply:
Ah you're right, I don't need the promise at all!